draft-ietf-tls-esni.md

```diff
@@ -699,24 +699,30 @@ MAY offer to resume sessions established without ECH.
 ## Client-Facing Server
 
 Upon receiving an "encrypted_client_hello" extension, the client-facing server
-MUST check that it is able to negotiate TLS 1.3 or greater. If not, it MUST
-abort the connection with a "handshake_failure" alert.
+first checks that it is able to negotiate TLS 1.3 or higher. If not, then it
+MUST ignore the extension and proceed with the handshake as usual. Next, the
+server checks that the ClientHello contains a "server_name" extension and that
+the indicated name is the public name of at least one of its ECH configurations.
+If not, then the server MUST ignore the "encrypted_client_hello" extension and
+proceed as usual. (Presumably the extension was generated as specified in
```

Comment:
Seems this new text is redundant with the text right below it. "If the ClientECH value does not match any known ECHConfig structure, it [...]".

Reply:
The public name not matching is not the same as the ClientECH not matching a known ECHConfig. In particular, the former does not imply the latter.